### PR TITLE
Switch actix-web to rustls

### PR DESCRIPTION
**Pull request by CodeSandwich**
This is first step to upgrading actix-web. In 1.0 it dropped support for native-tls and kept only openssl and rustls.

### CODE REVIEW

File: doc/quickstart/02_passive_node.md [hunk 1]
```diff
@@ -62,7 +62,9 @@ Description of the fields:
       - `file`: path to the log file.
 - `rest`: (optional) Configuration of the REST endpoint.
     - `listen`: _address_:_port_ to listen for requests
-    - `pkcs12`: (optional) Certificate file
+    - `tls`: (optional) enables TLS and disables plain HTTP if provided
+      - `server_certs_file`: path to server X.509 certificate chain file, must be PEM-encoded and contain at least 1 item
```
Comment (mzabaluev) [inline doc/quickstart/02_passive_node.md]: Could do away with the `server_` prefix (we don't have a REST client side in jormungandr), and maybe bikeshed it as `cert_file` if it's going to carry just one self-signed certificate most of the time; that will be more consistent with the `*CertFile*` error variants.

**Comment by NicolasDP** [comment]
Not to merge yet as we are preparing `master` for a release today

File: jormungandr/src/rest/server/error.rs [hunk 2]
```diff
@@ -1,38 +1,28 @@
...
+use thiserror::Error;
 
-#[derive(Debug)]
+#[derive(Debug, Error)]
 pub enum Error {
-    Pkcs12LoadFailed(IoError),
-    Pkcs12Invalid(TlsError),
-    BindFailed(IoError),
+    #[error("failed to open server certificate file")]
+    ServerCertFileOpenFailed(#[source] io::Error),
```
Comment (mzabaluev) [inline jormungandr/src/rest/server/error.rs]: All these `Server` prefixes just add typing, please remove them.

The existing members (`ServerAlreadyStopped` and below) do make some narrative sense.

File: jormungandr/src/rest/server/mod.rs [hunk 3]
```diff
@@ -37,23 +38,25 @@ pub struct ServerStopper {
 
 impl Server {
     pub fn start<F, H>(
-        pkcs12: Option<PathBuf>,
+        tls_config_opt: Option<TlsConfig>,
```
Comment (mzabaluev) [inline jormungandr/src/rest/server/mod.rs]: Don't need the `_opt` suffix.

**Pull request merged**